### PR TITLE
Twitter Content Provider now use Twitter's 1.1 Api.

### LINKED ISSUE #884: Need to update JabbR to use Twitter API 1.1 API
Previously, if you pasted a tweet into any room in JabbR, it'll post the tweet inline.

But now that twitter has deprecated the 1.0 API, you get a bunch of javascript errors.

So if you paste the API call into your browser, you'll get the following message returned:
{"errors": [{"message": "The Twitter REST API v1 is no longer active. Please migrate to API v1.1. https://dev.twitter.com/docs/api/1.1/overview.", "code": 68}]}

The twitter 1.0 API was officially retired on June 11th - https://dev.twitter.com/blog/api-v1-retirement-final-dates

### PR DESCRIPTION
**Pull request by PureKrome**
- Tweets are now embedded with via the statuses/oembed resource.
- Fixes #884.

### CODE REVIEW

File: JabbR/Chat.twitter.js [hunk 1]
```diff
@@ -4,21 +4,19 @@
     window.addTweet = function (tweet) {
         // Keep track of whether we're need the end, so we can auto-scroll once the tweet is added.
         var nearEnd = ui.isNearTheEnd(),
-            elements = null;
+            elements = null,
+            //id = tweet.url.replace(/[/]/g, '_');
```
Comment (davidfowl) [inline JabbR/Chat.twitter.js]: dead code

File: JabbR/Views/Home/index.cshtml [hunk 2]
```diff
@@ -151,12 +151,9 @@
     </script>
     <!-- TweetContentProvider: Should be extracted out if other content providers need templates -->
     <script id="tweet-template" type="text/x-jquery-tmpl">
-        <div class="user">
```
Comment (davidfowl) [inline JabbR/Views/Home/index.cshtml:154]: Are all of the other properties gone?

File: JabbR/Chat.twitter.js (hunk 1 above)
Comment (PureKrome) [inline JabbR/Chat.twitter.js]: shit soz. yep, that was my first attempt. will clean that.

File: JabbR/Views/Home/index.cshtml (hunk 2 above)
Comment (PureKrome) [inline JabbR/Views/Home/index.cshtml:154]: yep. Twitter provides the html code now, we don't have to generate it.

Comment (davidfowl) [inline JabbR/Views/Home/index.cshtml:154]: how do we make sure the styles are right? All inline css?

Comment (PureKrome) [inline JabbR/Views/Home/index.cshtml:154]: The HTML provided by Twitter includes a script to load their js for stuff. I guess it also includes their CSS?

[here's the doc with example output](https://dev.twitter.com/docs/api/1.1/get/statuses/oembed).

[Sample json result](https://api.twitter.com/1.1/statuses/oembed.json?id=240192632003911681)

and that json code looks like ..

```
{"cache_age":"3153600000","url":"https:\/\/twitter.com\/twitter\/statuses\/99530515043983360","height":null,"provider_url":"https:\/\/twitter.com","provider_name":"Twitter","author_name":"Twitter","version":"1.1","author_url":"https:\/\/twitter.com\/twitter","type":"rich","html":"\u003cblockquote class=\"twitter-tweet\"\u003e\u003cp\u003eCool! \u201c\u003ca href=\"https:\/\/twitter.com\/tw1tt3rart\"\u003e@tw1tt3rart\u003c\/a\u003e: \u003ca href=\"https:\/\/twitter.com\/search?q=%23TWITTERART&amp;src=hash\"\u003e#TWITTERART\u003c\/a\u003e \u2571\u2571\u2571\u2571\u2571\u2571\u2571\u2571 \u2571\u2571\u256d\u2501\u2501\u2501\u2501\u256e\u2571\u2571\u256d\u2501\u2501\u2501\u2501\u256e \u2571\u2571\u2503\u2587\u2506\u2506\u2587\u2503\u2571\u256d\u252b\u24e6\u24d4\u24d4\u24da\u2503 \u2571\u2571\u2503\u25bd\u25bd\u25bd\u25bd\u2503\u2501\u256f\u2503\u2661\u24d4\u24dd\u24d3\u2503 \u2571\u256d\u252b\u25b3\u25b3\u25b3\u25b3\u2523\u256e\u2571\u2570\u2501\u2501\u2501\u2501\u256f \u2571\u2503\u2503\u2506\u2506\u2506\u2506\u2503\u2503\u2571\u2571\u2571\u2571\u2571\u2571 \u2571\u2517\u252b\u2506\u250f\u2513\u2506\u2523\u251b\u2571\u2571\u2571\u2571\u2571\u201d\u003c\/p\u003e&mdash; Twitter (@twitter) \u003ca href=\"https:\/\/twitter.com\/twitter\/statuses\/99530515043983360\"\u003eAugust 5, 2011\u003c\/a\u003e\u003c\/blockquote\u003e\n\u003cscript async src=\"\/\/platform.twitter.com\/widgets.js\" charset=\"utf-8\"\u003e\u003c\/script\u003e","width":550}
```

so yeah.. they give us the html and i just plop it in.

**Comment by PureKrome** [comment]
Pulling PR temporarily. got another idea.

**Comment by davidfowl** [comment]
PR on the wrong branch,

**Comment by PureKrome** [comment]
what branch should it be against?

**Comment by davidfowl** [comment]
https://github.com/JabbR/JabbR/blob/master/CONTRIBUTING.md

**Comment by PureKrome** [comment]
ta. donno how my previous PR's slipped in :smile: Got it now!

**Pull request closed without merging**